changes/2020-06-17_encrypt-max-plaintext-length-input/change.md

```diff
@@ -0,0 +1,182 @@
+If there were such a case,
+then such a CMM is not correctly using the intent behind `maxPlaintextLength`.
+It is better to restrict flexibilty here so that users have less opportunity
+to depend on behaviors from CMMs that are misusing this value.
+
+## Drawbacks
+
+This change does not prescribe the name of this input in implementations,
+and thus there will not be a standardized name or way of specifying this input across implementations.
+This is because this control already exists in implementations under different names,
+and it is not worth breaking all customers to rename this input
+without significantly changing this input to benefit the customer.
+
+## Security Implications
+
+This change SHOULD NOT have any security implications.
+
+## Operational Implications
+
+This change will break any JS user that is currently using Encrypt with a known-length plaintext
+and inputting a plaintextLength less than that length.
+
+This change will break any Python user that depends on the source_length being sent to the
+CMM instead of the true length of a known-length plaintext.
+
+## Guide-level Explanation
+
+This change renames the `plaintext length` Encryption input to `max plaintext length` in the specification,
+and clarifies its behavior in the Encrypt operation.
+
+If during the Encrypt operation it is determined that
+the actual plaintext length is greater than a supplied `max plaintext length`,
+the operation MUST fail.
+
+Additionally, when calling the CMM's GetEncryptionMaterials call during the Encrypt operation,
+the operation MUST pass in the length of the input plaintext if it can be known.
+If it can't be known, the operation passes in `max plaintext input` if supplied.
+If it can't be known and no `max plaintext input` is supplied, the operation supplies no value for
+this field.
+
+This change also renames the `plaintext length` field in the CMM interface to `max plaintext length`.
+
+Finally, this change specifies when this input should be available for users to specify:
+if and only if the input plaintext can be of unknown-length.
+
+## Reference-level Explanation
+
+We need to update the spec to make the purpose and behavior of the `plaintext length` input clear
+by renaming it and specifying its exact behavior within the Encrypt operation:
+
+When performing the encrypt operation on a unknown-length plaintext,
+users MUST be able to specify an optional parameter `maxPlaintextLength`.
+The value of this field represents the bound on the length of the plaintext to be encrypted.
+The ESDK MUST NOT encrypt a plaintext greater than this length,
+and MUST fail if it can be determined during encryption that the actual plaintext length
+is greater than what the user supplied on input.
+The actual name of this input, and how the user specifies this value for the Encrypt operation
+MAY be different per implementation.
+
+We also need to specify the exact behavior for how this input is used in the CMM's GetEncryptionMaterials call:
```

Comment:
```suggestion
We also need to specify the exact behavior for how this input is passed to the CMM's GetEncryptionMaterials call:
```

I don't think you mean "used" because you don't talk about how the CMM is using the value.